src/components/m-datepicker-body/index.js

```diff
@@ -17,58 +17,153 @@ class AXADatepickerBody extends BaseComponentGlobal {
     locale: localePropType,
     value: PropTypes.string,
     index: PropTypes.number,
+    allowedYears: PropTypes.arrayOf(PropTypes.number),
     year: PropTypes.number,
     month: PropTypes.number,
     day: PropTypes.number,
-    allowedYears: PropTypes.arrayOf(PropTypes.number),
+    date: PropTypes.string,
+    cells: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  static get observedAttributes() {
+    return ['day', 'month', 'year', 'cells', 'value', 'index', 'date'];
   }
 
   init() {
     super.init({ styles, template });
-
-    this.datepickerBody = new DatepickerBody(this);
-    // does this provide context (See docs for context) ?
-    // this.provideContext()
-
-    // or do you want to consume a specific context
-    // this.consumeContext('axa-context-provider');
   }
 
-  /**
-   * REF: https://www.w3.org/TR/custom-elements/#custom-element-conformance
-   */
   connectedCallback() {
     super.connectedCallback();
-
     this.className = `${this.initialClassName} m-datepicker-body`;
-    // Your DOM interaction here, but keep it decoupled.
-    // If you don't have any, just remove this function
+
+    const date = new Date(this.props.year, this.props.month - 1, this.props.day);
+    this.store = new Store(this.locale, date);
+
+    this.onDatepickerBodyCellClick = on(
+      this, EVENTS.CLICK, 'js-datepicker__calender-body__cell',
+      e => this.handleDatepickerBodyCellClick(e), { capture: true, passive: false },
+    );
+    // Set Cells
+    this.props.cells = this.store.cells;
   }
 
-  disconnectedCallback() {
-    super.disconnectedCallback();
+  didRenderCallback() {
+    // this is not working in connectedCallback...
+    const date = new Date(this.props.year, this.props.month - 1, this.props.day);
+    if (!this.getAttribute('date')) {
+      this.setAttribute('date', date.toISOString());
+    }
+  }
+
+  handleDatepickerBodyCellClick(e) {
+    e.preventDefault();
+    const index = parseInt(e.target.dataset.index, 10);
+
+    // We parse the iso date to work with
+    const date = new Date(Date.parse(e.target.dataset.value));
+    this.year = date.getFullYear();
+    this.month = date.getMonth();
+    this.day = date.getDate();
+    this.index = index;
+
+    // Set the day to the chosen day
+    this.date = date.toISOString();
+  }
+
+  get index() {
```

Comment:
@LucaMele 
@jackblackCH 
The boilerplate for getters/setters repeat most of the time.
May we find a more generic approach to add them

Reply:
@AndyOGo agree. But is also part of best practice from google. https://developers.google.com/web/fundamentals/web-components/best-practices 

So we should maybe generate them in future automatically or something like this. But i think for this PR is ok to merge or not?

Reply:
@LucaMele 
Yep I know that best-practise guide.
Anyway, we do not follow much of it right now (like we would need to use shadow-DOM, no complex data types as attributes, etc.)

Reply:
yepp lets leave it as @jackblackCH did and issue this in a separate task

Reply:
Sure, let's discuss it in a code review.